rfcs/0001-messages.md

```diff
@@ -0,0 +1,422 @@
+    Dealer,
+    /// The fixed participant ID for the aggregator.
+    Aggregator,
+}
+
+/// The fixed participant ID for the dealer.
+const DEALER_PARTICIPANT_ID: u64 = u64::MAX - 1;
+
+/// The fixed participant ID for the aggregator.
+const AGGREGATOR_PARTICIPANT_ID: u64 = u64::MAX;
+
+/// The maximum `ParticipantId::Signer` in this serialization format.
+///
+/// We reserve two participant IDs for the dealer and aggregator.
+const MAX_SIGNER_PARTICIPANT_ID: u64 = u64::MAX - 2;
+```
+
+### Payloads
+
+Each payload defines a new message:
+
+```rust
+/// The data required to serialize `frost::SharePackage`.
+///
+/// The dealer sends this message to each signer for this round.
+/// With this, the signer should be able to build a `SharePackage` and use
+/// the `sign()` function.
+///
+/// Note: `frost::SharePackage.public` can be calculated from `secret_share`.
+struct messages::SharePackage {
+    /// The public signing key that represents the entire group:
+    /// `frost::SharePackage.group_public`.
+    group_public: VerificationKey<SpendAuth>,
+    /// This participant's secret key share: `frost::SharePackage.share.value`.
+    secret_share: frost::Secret,
+    /// The commitments to the coefficients for our secret polynomial _f_,
+    /// used to generate participants' key shares. Participants use these to perform
+    /// verifiable secret sharing.
+    share_commitment: Vec<frost::Commitment>,
+}
+
+/// The data required to serialize `frost::SigningCommitments`.
+///
+/// Each signer must send this message to the aggregator.
+/// A signing commitment from the first round of the signing protocol.
+struct messages::SigningCommitments {
+    /// The hiding point: `frost::SigningCommitments.hiding`
+    hiding: frost::Commitment,
+    /// The binding point: `frost::SigningCommitments.binding`
+    binding: frost::Commitment,
+}
+
+/// The data required to serialize `frost::SigningPackage`.
+///
+/// The aggregator decides what message is going to be signed and
+/// sends it to each signer with all the commitments collected.
+struct messages::SigningPackage {
+    /// The collected commitments for each signer as a hashmap of
+    /// unique participant identifiers: `frost::SigningPackage.signing_commitments`
+    ///
+    /// Signing packages that contain duplicate or missing `ParticipantId`s are invalid.
+    signing_commitments: HashMap<ParticipantId, SigningCommitments>,
+    /// The message to be signed: `frost::SigningPackage.message`.
+    ///
+    /// Each signer should perform protocol-specific verification on the message.
+    message: Vec<u8>,
+}
+
+/// The data required to serialize `frost::SignatureShare`.
+///
+/// Each signer sends their signatures to the aggregator who is going to collect them
+/// and generate a final spend signature.
+struct messages::SignatureShare {
+     /// This participant's signature over the message: `frost::SignatureShare.signature`
+    signature: frost::Scalar,
```

Comment:
It will be tricky to change this `Scalar` to a frost type as the right one will be `SignatureShare` which repeats the index id:

https://github.com/ZcashFoundation/redjubjub/blob/main/src/frost.rs#L397

Reply:
I added a new type in FROST: https://github.com/ZcashFoundation/redjubjub/pull/107 and applied it to this spec at https://github.com/ZcashFoundation/redjubjub/pull/85/commits/189efc20845d02723996abedbcdb3f4857e65dcb